### PR TITLE
KV: `VALIDATION_ERROR` is not an internal error as it can also be caused by pruned contracts [KVL-1207]

### PR DESCRIPTION
**Pull request by fabiotudone-da**
CHANGELOG_BEGIN
[Integration Kit] `VALIDATION_ERROR` is not an internal error anymore as it can also be caused by pruned contracts
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:118]:
```suggestion
        logger.warn(
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]:
```suggestion
      // Missing package or contract not specified as input
      case missingInputErr @ Err.MissingInputState(key) => 
```

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (SamirTalwar-DA) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:128]:
```suggestion
        logger.warn(
```

Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:128]: This is actually an internal error, i.e. the participant really did something wrong. Why are you suggesting to log it at `warn`?

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala]: ae9a5dc

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: Oh, my apologies; I thought this could also be triggered by pruned contracts. Ignore me.

**Review by SamirTalwar-DA — APPROVED** [review]

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (cocreature) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:115]: Should we be treating packages and contracts the same here? Afaict a missing package really is an issue in the participant right now whereas a missing contract can arise from committer pruning.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (cocreature) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: I’m confused, why can this not be triggered by pruning?

Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: In this logic, it looks to me like this can be thrown only [here](https://github.com/digital-asset/daml/blob/main/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala#L56), i.e. if the participant doesn't declare a needed input. I don't see how this can be caused by committer pruning but maybe I am missing something, am I?

Comment (hubert-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: This is not something we were able to reproduce. The error we get from PKV reference architecture conformance tests:
```
30-11-2021 11:26:00.624 UTC|ERROR||c.d.l.p.s.k.KeyValueConsumption|in-memory-ledger-akka.actor.default-dispatcher-16|||||VALIDATION_FAILURE(4,RWArchiv): Validation failure: Contract could not be found with id 00e471ec59434402166e631f1a54e84f01db9073ad38c8c32c4d8ba4e1f4789a44|{participantId: "conformance-test", offset: "01000000000006ea0000000000000000", err-context: "{location=TransactionRejections.scala:544}"}|
```
comes from here: https://github.com/digital-asset/daml/pull/12108/files#diff-7cde65905babc148f1f6872cdc5c034ae66d0390e09d4f764c47910ae8b5d2d8R108

This PR seems to be sufficient to fix this particular issue.

**Review by hubert-da — APPROVED** [review]

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (hubert-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: Sorry @fabiotudone-da, didn't refresh the page and missed your comment

Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: @hubert-da I forgot to add that it indeed does, already tested with a local SDK build.

Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: @hubert-da It shouldn't be in this particular case, as [that](https://github.com/digital-asset/daml/blob/main/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala#L56) is a validation on the declared transaction inputs.

Comment (hubert-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: Ok, makes sense, thanks.

Comment (cocreature) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:117]: thanks for the explanation, sounds good

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala
Comment (fabiotudone-da) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/CommitterModelConformanceValidator.scala:115]: Discussed above

**Review by cocreature — APPROVED** [review]
LGTM thank you!

**Comment by fabiotudone-da** [comment]
@meiersi-da Would you like to review before merge or can I go ahead?

**Pull request merged**
